table/tables/tables.go

```diff
@@ -388,6 +388,10 @@ func (t *tableCommon) rebuildIndices(ctx sessionctx.Context, rm kv.RetrieverMuta
 			untouched = false
 			break
 		}
+		// If txn is auto commit and index is untouched, no need to write index value.
+		if untouched && !ctx.GetSessionVars().InTxn() {
+			return nil
```

Comment:
why return nil here? the following indices may need to be created.

Reply:
The untouched index value must already exist in tikv, no need to create again.